Slim/Control/Queries.pm

```diff
@@ -4281,6 +4285,25 @@ sub statusQuery {
 									$data, $tags,
 									'playlist index', $idx, $fast
 								);
+
+						if ( $tags =~ /2/ ) {
+							# build a hash containing the grouping data for each album instance/track in the play queue.
+							my $track = $request->{_results}->{'playlist_loop'}[$count];
+							my $album = $track->{'album_id'};
+							my $tracknum = $track->{'tracknum'};
+							my $albumTrack = $album*10000+$tracknum; # used to detect a new album or a repeat of tracks from the same album
+							my $work = $track->{'work_id'} || '';
+							my $grouping = $track->{'grouping'} || '';
+							my $performance = $track->{'performance'} || '';
+							my $group = "$work##$grouping##$performance";
```

Comment:
Please do not define variables which are only used once. Just access the raw value when needed. IMHO it's harder to read lengthy code caused by variable assignments than reading slightly less visually pleasing expressions with lengthy value references.